include/nigiri/routing/search.h

```diff
@@ -136,17 +138,29 @@ struct search {
     state_.starts_.clear();
     add_start_labels(q_.start_time_, true);
 
+    auto processing_start_time = std::chrono::system_clock::now();
+    auto timeout_reached = [=, this]() -> bool {
```

Comment:
```suggestion
    auto const timeout_reached = [&]() -> bool {
```

As it doesn't change.

Capturing everything by `&` is usually the right thing to do. The compiler eliminates everything that's not used in the lambda. So no worries about copying a bunch of pointers around.